### ISSUE TITLE
NVidia Drivers not compile. Kernel 4.14.16.

### ISSUE THREAD

**Issue by romhdd** [Thinstation/thinstation#309]
Thinstation 5.6.
I  building thinstation 5.6. during compilation NVidia Drivers halted on error:
"This driver requires the ability to change memory types!"
Tested on dif. linux and come to conclusion.
not compiling after installation of linux-kernel-4.13 and above...
Thinstation 5.6 - kernel 4.14
Need help

**Comment by Doncuppjr** [comment]
Spectre and meltdown mitigation’s. Try a newer driver.


Sent from Yahoo Mail for iPhone


On Friday, May 18, 2018, 3:31 AM, romhdd <notifications@github.com> wrote:


Thinstation 5.6.
I building thinstation 5.6. during compilation NVidia Drivers halted on error:
"This driver requires the ability to change memory types!"
Tested on dif. linux and come to conclusion.
not compiling after installation of linux-kernel-4.13 and above...
Thinstation 5.6 - kernel 4.14
Need help

**Comment by romhdd** [comment]
latest driver 340.106 
the same issue

**Comment by romhdd** [comment]
ERROR: Kernel configuration is invalid."
include/generated/autoconf.h or include/config/auto.conf are missing."

**Comment by romhdd** [comment]
Started with builded image:
[    12.057] (II) LoadModule: "nvidia"
[    12.057] (II) Loading /usr/lib/xorg/modules/drivers/nvidia_drv.so
[    12.057] (II) Module nvidia: vendor="NVIDIA Corporation"
[    12.057]    compiled for 4.0.2, module version = 1.0.0
[    12.057]    Module class: X.Org Video Driver
[    12.057] (II) NVIDIA dlloader X Driver  340.106  Tue Jan  9 14:11:04 PST 2018
[    12.057] (II) NVIDIA Unified Driver for all Supported NVIDIA GPUs
[    12.057] (++) using VT number 5

[    12.114] (EE) NVIDIA: Failed to initialize the NVIDIA kernel module. Please see the
[    12.114] (EE) NVIDIA:     system's kernel log for additional error messages and
[    12.114] (EE) NVIDIA:     consult the NVIDIA README for details.
[    12.115] (EE) No devices detected.
[    12.115] (EE)
Fatal server error:
[    12.115] (EE) no screens found(EE)
[    12.115] (EE)
Please consult the ThinStation made by Don Cupp support
I consult   :-)

**Comment by Doncuppjr** [comment]
Contents of messages?


Sent from Yahoo Mail for iPhone


On Friday, May 18, 2018, 4:30 AM, romhdd <notifications@github.com> wrote:


Started with builded image:
[ 12.057] (II) LoadModule: "nvidia"
[ 12.057] (II) Loading /usr/lib/xorg/modules/drivers/nvidia_drv.so
[ 12.057] (II) Module nvidia: vendor="NVIDIA Corporation"
[ 12.057] compiled for 4.0.2, module version = 1.0.0
[ 12.057] Module class: X.Org Video Driver
[ 12.057] (II) NVIDIA dlloader X Driver 340.106 Tue Jan 9 14:11:04 PST 2018
[ 12.057] (II) NVIDIA Unified Driver for all Supported NVIDIA GPUs
[ 12.057] (++) using VT number 5

[ 12.114] (EE) NVIDIA: Failed to initialize the NVIDIA kernel module. Please see the
[ 12.114] (EE) NVIDIA: system's kernel log for additional error messages and
[ 12.114] (EE) NVIDIA: consult the NVIDIA README for details.
[ 12.115] (EE) No devices detected.
[ 12.115] (EE)
Fatal server error:
[ 12.115] (EE) no screens found(EE)
[ 12.115] (EE)
Please consult the ThinStation made by Don Cupp support
I consult :-)

**Comment by romhdd** [comment]
when compiling NVIDIA Drivers v.340.106 an error is received:
```
ERROR: Kernel configuration is invalid."
include/generated/autoconf.h or include/config/auto.conf are missing."
Run 'make oldconfig && make prepare' on kernel src to fix it
```
Building passed.
I'm booting from the this image and see black screen
I`m connect to terminal through ssh and viewing Xorg.log

> ...
> [ 12.057] (II) LoadModule: "nvidia"
> [ 12.057] (II) Loading /usr/lib/xorg/modules/drivers/nvidia_drv.so
> [ 12.057] (II) Module nvidia: vendor="NVIDIA Corporation"
> [ 12.057] compiled for 4.0.2, module version = 1.0.0
> [ 12.057] Module class: X.Org Video Driver
> [ 12.057] (II) NVIDIA dlloader X Driver 340.106 Tue Jan 9 14:11:04 PST 2018
> [ 12.057] (II) NVIDIA Unified Driver for all Supported NVIDIA GPUs
> [ 12.057] (++) using VT number 5
> ....
> [ 12.114] (EE) NVIDIA: Failed to initialize the NVIDIA kernel module. Please see the
> [ 12.114] (EE) NVIDIA: system's kernel log for additional error messages and
> [ 12.114] (EE) NVIDIA: consult the NVIDIA README for details.
> [ 12.115] (EE) No devices detected.
> [ 12.115] (EE)
> Fatal server error:
> [ 12.115] (EE) no screens found(EE)
> [ 12.115] (EE)
> ....

**Comment by Doncuppjr** [comment]
/var/log/messages


Sent from Yahoo Mail for iPhone


On Friday, May 18, 2018, 5:10 AM, romhdd <notifications@github.com> wrote:


when compiling NVIDIA Drivers v.340.106 an error is received:
ERROR: Kernel configuration is invalid."
include/generated/autoconf.h or include/config/auto.conf are missing."
Run 'make oldconfig && make prepare' on kernel src to fix it

Building passed.
I'm booting from the this image and see black screen
I`m connect to terminal through ssh and viewing Xorg.log


...
[ 12.057] (II) LoadModule: "nvidia"
[ 12.057] (II) Loading /usr/lib/xorg/modules/drivers/nvidia_drv.so
[ 12.057] (II) Module nvidia: vendor="NVIDIA Corporation"
[ 12.057] compiled for 4.0.2, module version = 1.0.0
[ 12.057] Module class: X.Org Video Driver
[ 12.057] (II) NVIDIA dlloader X Driver 340.106 Tue Jan 9 14:11:04 PST 2018
[ 12.057] (II) NVIDIA Unified Driver for all Supported NVIDIA GPUs
[ 12.057] (++) using VT number 5
....
[ 12.114] (EE) NVIDIA: Failed to initialize the NVIDIA kernel module. Please see the
[ 12.114] (EE) NVIDIA: system's kernel log for additional error messages and
[ 12.114] (EE) NVIDIA: consult the NVIDIA README for details.
[ 12.115] (EE) No devices detected.
[ 12.115] (EE)
Fatal server error:
[ 12.115] (EE) no screens found(EE)
[ 12.115] (EE)
....

**Comment by Doncuppjr** [comment]
Everything should compile in spite of that. I think it’s module signing.


Sent from Yahoo Mail for iPhone


On Friday, May 18, 2018, 5:12 AM, romhdd <notifications@github.com> wrote:


Following the recommendations NVidia Installation:
ERROR: Kernel configuration is invalid."
include/generated/autoconf.h or include/config/auto.conf are missing."
Run 'make oldconfig && make prepare' on kernel src to fix it

cd /usr/src/kernels/4.14.16TS_SMP/
make oldconfig
make prepare
make[1]: *** No rule to make target 'arch/x86/entry/syscalls/syscall_32.tbl', needed by 'arch/x86/include/generated/asm/syscalls_32.h'.  Stop.
arch/x86/Makefile:253: recipe for target 'archheaders' failed
make: *** [archheaders] Error 2

**Comment by Thinstation** [comment]
Yeah, signatures are enforced for kernel modules. You need to sign them with the kernel keys that are provided by default, or make your own keys, recompile the kernel and nvidia module.

**Comment by Thinstation** [comment]
I think this is now fixed in 5.6-Stable, can you please test and report?

**Comment by romhdd** [comment]
Ok. I will do it in a few days.

**Comment by romhdd** [comment]
the same thing

**Comment by Doncuppjr** [comment]
You pulled from git?


Sent from Yahoo Mail for iPhone


On Thursday, July 19, 2018, 7:10 AM, romhdd <notifications@github.com> wrote:


the same thing

**Comment by romhdd** [comment]
wget "https://github.com/Thinstation/thinstation/archive/5.6-Stable.zip"

**Comment by Doncuppjr** [comment]
I have not updated the release, only git. Please clone lastest 5.6-Stable


Sent from Yahoo Mail for iPhone


On Thursday, July 19, 2018, 7:16 AM, romhdd <notifications@github.com> wrote:


wget "https://github.com/Thinstation/thinstation/archive/5.6-Stable.zip"

**Comment by Doncuppjr** [comment]
Git Cline —depth 1 Thinstation/thinstation 
  
|  
|   
|   
|   |    |

   |

  |
|  
|    |  
Thinstation/thinstation
 
thinstation - A framework for making thin and light Linux based images for x86 based machines and thinclients.
  |   |

  |

  |

 
-b 5.6-Stable


Sent from Yahoo Mail for iPhone


On Thursday, July 19, 2018, 7:16 AM, romhdd <notifications@github.com> wrote:


wget "https://github.com/Thinstation/thinstation/archive/5.6-Stable.zip"

**Comment by romhdd** [comment]
ok

**Comment by romhdd** [comment]
git clone --depth 1 git://github.com/Thinstation/thinstation.git -b 5.6-Stable

**Comment by Thinstation** [comment]
Dang it, that's what I get for taking code. The kernel build process was updated, and the signing portion was partially broken so that only if you compiled the kernel, and then out-of-tree modules without cleaning, would it all work. I think I have it fixed.

**Comment by romhdd** [comment]
doesn't work

**Comment by Doncuppjr** [comment]
Yeah, I’m still working through all the issues with Nvidia. Fixed module signing, but now I get a garbled screen, and no errors. Gonna try a different card, but I’m waiting for it to arrive.


Sent from Yahoo Mail for iPhone


On Tuesday, July 24, 2018, 12:16 AM, romhdd <notifications@github.com> wrote:


doesn't work

**Comment by Doncuppjr** [comment]
All fixed in 6.1-Stable. 
It looks like NVidia is dropping support for 32 bit linux, because I can't backport to 5.6-Stable.
    On Tuesday, July 24, 2018, 7:06:52 AM PDT, Don Cupp <doncuppjr@yahoo.com> wrote:  
 
 Yeah, I’m still working through all the issues with Nvidia. Fixed module signing, but now I get a garbled screen, and no errors. Gonna try a different card, but I’m waiting for it to arrive.


Sent from Yahoo Mail for iPhone


On Tuesday, July 24, 2018, 12:16 AM, romhdd <notifications@github.com> wrote:


doesn't work
